File: BEmuJava/bemu/src/main/java/com/bloomberglp/blpapi/MarketElementServiceStatus.java
//------------------------------------------------------------------------------
// <copyright project="BEmu_maven" file="/BEmu_maven/bemu/src/main/java/com/bloomberglp/blpapi/MarketElementServiceStatus.java" company="Jordan Robinson">
//     Copyright (c) 2013 Jordan Robinson. All rights reserved.
//
//     The use of this software is governed by the Microsoft Public License
//     which is included with this distribution.
// </copyright>
//------------------------------------------------------------------------------

package com.bloomberglp.blpapi;

import com.bloomberglp.blpapi.Element;
import com.bloomberglp.blpapi.Name;
import com.bloomberglp.blpapi.Schema;

public class MarketElementServiceStatus extends Element
{
    private final MarketElementString _serviceName;

    MarketElementServiceStatus(MarketMessageServiceStatus arg) throws Exception
    {
        this._serviceName = (MarketElementString)arg.getElement("serviceName");
    }
	
    public Name name() throws Exception
    {
    	return new Name("ServiceOpened");
    }
    
    public int numValues()
    {
    	return 1;
    }
    
    public int numElements()
    {
    	return 1;
    }
    
    public boolean isComplexType()
    {
    	return true;
    }
    
    public boolean isArray()
    {
    	return false;
    }
    
    public boolean isNull()
    {
    	return false;
    }

    public Schema.Datatype datatype()
    {
    	return Schema.Datatype.SEQUENCE;
    }
    
    protected StringBuilder prettyPrint(int tabIndent) throws Exception
    {
        StringBuilder result = new StringBuilder();

        result.append(String.format("%s = {%s", this.name().toString(), System.getProperty("line.separator")));
        result.append(this._serviceName.prettyPrint(tabIndent + 1));
        result.append(String.format("}", System.getProperty("line.separator")));

        return result;
    }
    
}

File: BEmu/MarketDataRequest/MarketElementServiceStatus.cs
//------------------------------------------------------------------------------
// <copyright project="BEmu_csh" file="MarketDataRequest/MarketElementServiceStatus.cs" company="Jordan Robinson">
//     Copyright (c) 2013 Jordan Robinson. All rights reserved.
//
//     The use of this software is governed by the Microsoft Public License
//     which is included with this distribution.
// </copyright>
//------------------------------------------------------------------------------

namespace Bloomberglp.Blpapi.MarketDataRequest
{
    using System;
    using System.Collections.Generic;
    using System.Linq;
    using System.Text;

    internal class MarketElementServiceStatus : Element
    {
        private readonly MarketElementString _serviceName;

        internal MarketElementServiceStatus(MarketMessageServiceStatus arg)
        {
            this._serviceName = (MarketElementString)arg.Elements.First();
        }

        public override int NumElements { get { return 1; } }
        public override int NumValues { get { return 1; } }
        public override Name Name { get { return new Name("ServiceOpened"); } }
        public override bool IsArray { get { return false; } }
        public override bool IsComplexType { get { return true; } }
        public override bool IsNull { get { return false; } }
        public override object this[int index] { get { return null; } }
        public override Schema.Datatype Datatype { get { return Schema.Datatype.SEQUENCE; } }

        

        public override Element GetElement(int index)
        {
            if (index == 0)
                return this._serviceName;
            else
                return base.GetElement(index);
        }

        internal override StringBuilder PrettyPrint(int tabIndent, bool surroundValueWithQuotes = false)
        {
            StringBuilder result = new StringBuilder();
            result.AppendFormat("{0} = {{{1}", this.Name.ToString(), Environment.NewLine);
            result.Append(this._serviceName.PrettyPrint(1));
            result.Append("}" + Environment.NewLine);
            return result;
        }
    }
}
